sql/insert.go

```diff
@@ -93,24 +115,27 @@ func (p *planner) Insert(n *parser.Insert) (planNode, error) {
 		return nil, err
 	}
 	if err := p.db.Run(&b); err != nil {
+		if tErr, ok := err.(*proto.ConditionFailedError); ok {
+			return nil, fmt.Errorf("duplicate key value %q violates unique constraint %s", tErr.ActualValue.Bytes, "TODO(tamird)")
+		}
```

Comment:
Discussed this with @vivekmenezes today - we can't just instantiate a transaction here since this statement can execute as part of a SQL transaction. In other words, the transaction has to be passed into this method.

I filed https://github.com/cockroachdb/cockroach/issues/1889 earlier today to track.